meson.build

```
@@ -0,0 +1,533 @@
+        cfg_var.set('UNTAGGED_EXIF', '1')
+    endif
+endif
+
+
+libjpeg_dep = dependency('libjpeg', required: get_option('jpeg'))
+if libjpeg_dep.found()
+    libvips_deps += libjpeg_dep
+    cfg_var.set('HAVE_JPEG', '1')
+    # features like trellis quant are exposed as extension parameters ...
+    # mozjpeg 3.2 and later have #define JPEG_C_PARAM_SUPPORTED, but we must 
+    # work with earlier versions
+    if cc.has_function('jpeg_c_bool_param_supported', prefix: '#include <stdio.h>\n#include <jpeglib.h>', dependencies: libjpeg_dep)
+        cfg_var.set('HAVE_JPEG_EXT_PARAMS', '1')
+    endif
+endif
+
+# Look for libspng first
+# 0.6.1 uses "libspng.pc", git master libspng uses "spng.pc"
+# We can simplify this when requiring meson>=0.60.0
+spng_dep = dependency('libspng', version: '>=0.6', required: false)
+if not spng_dep.found()
+    spng_dep = dependency('spng', version: '>=0.6', required: get_option('spng'))
+endif
+if not get_option('spng').disabled() and spng_dep.found()
+    libvips_deps += spng_dep
+    cfg_var.set('HAVE_SPNG', '1')
+endif
+
+# we can have both PNG and SPNG enabled, with SPNG for read and PNG for write
+png_dep = dependency('libpng', version: '>=1.2.9', required: get_option('png'))
+if png_dep.found()
+    libvips_deps += png_dep
+    cfg_var.set('HAVE_PNG', '1')
+    if cc.has_function('png_set_chunk_malloc_max', prefix: '#include <png.h>', dependencies: png_dep)
+        cfg_var.set('HAVE_PNG_SET_CHUNK_MALLOC_MAX', '1')
+    endif
+endif
+
+
+# libwebp ... target 0.6+ to reduce complication
+# webp has the stuff for handling metadata in two separate libraries  -- we
+# insit on having all of them
+libwebp_dep = dependency('libwebp', version: '>=0.6', required: get_option('png'))
+if libwebp_dep.found()
+    libvips_deps += libwebp_dep
+    libvips_deps += dependency('libwebpmux', version: '>=0.6')
+    libvips_deps += dependency('libwebpdemux', version: '>=0.6')
+    cfg_var.set('HAVE_LIBWEBP', '1')
+endif
+
+pangocairo_dep = dependency('pangocairo', required: get_option('pangocairo'))
+if pangocairo_dep.found()
+    libvips_deps += pangocairo_dep
+    cfg_var.set('HAVE_PANGOCAIRO', '1')
+endif
+
+fontconfig_dep = dependency('fontconfig', required: get_option('fontconfig'))
+if fontconfig_dep.found() and pangocairo_dep.found()
+    libvips_deps += fontconfig_dep
+    cfg_var.set('HAVE_FONTCONFIG', '1')
+endif
+
+libtiff_dep = dependency('libtiff-4', required: get_option('tiff'))
+if libtiff_dep.found()
+    libvips_deps += libtiff_dep
+    cfg_var.set('HAVE_TIFF', '1')
+    # ZSTD and WEBP in TIFF added in libtiff 4.0.10
+    if cc.get_define('COMPRESSION_WEBP', prefix: '#include <tiff.h>', dependencies: libtiff_dep) != ''
+        cfg_var.set('HAVE_TIFF_COMPRESSION_WEBP', '1')
+    endif
+endif
+
+# 2.40.3 so we get the UNLIMITED open flag
+librsvg_dep = dependency('librsvg-2.0', version: '>=2.40.3', required: get_option('rsvg'))
+cairo_dep = dependency('cairo', version: '>=1.2', required: get_option('rsvg'))
```

Oh, it looks like I mistakenly assumed that `popplerload` didn't use cairo in one of my patches. Perhaps we could do something like this?:
```suggestion
cairo_dep = dependency('cairo', version: '>=1.2', required: get_option('rsvg') or get_option('poppler'))
```
(untested)

and change
```meson
elif libpoppler_dep.found()
```
below to
```meson
elif libpoppler_dep.found() and cairo_dep.found()
```

BTW, I suppose that it's safe to add the same dependency twice to `libvips_deps`? Since
```meson
        libvips_deps += libpoppler_dep
```
needs probably to be changed to:
```meson
        libvips_deps += libpoppler_dep
        libvips_deps += cairo_dep
```

---

It's safe to add the dependency several times :), changed